src/hipscat_import/soap/map_reduce.py

```diff
@@ -0,0 +1,105 @@
+"""Inner methods for SOAP"""
+
+import numpy as np
+import pandas as pd
+from hipscat.io import file_io
+from hipscat.io.paths import pixel_catalog_file
+
+
+def _count_joins_for_object(source_data, object_catalog_dir, object_id_column, object_pixel):
+    object_path = pixel_catalog_file(
+        catalog_base_dir=object_catalog_dir,
+        pixel_order=object_pixel.order,
+        pixel_number=object_pixel.pixel,
+    )
+    object_data = pd.read_parquet(path=object_path, columns=[object_id_column]).set_index(object_id_column)
```

Comment:
If we have, this could do with a wrapper method

Reply:
Max did just merge a method for this if you don't mind replacing the read_parquet? https://github.com/astronomy-commons/hipscat/pull/123

Reply:
Max actually made that method in a PR a few days ago. https://github.com/astronomy-commons/hipscat/pull/123

But it can wait for another PR to replace all the read_parquet